lib/teamsnap.rb

```diff
@@ -137,12 +154,21 @@ def init(opts = {})
     end
 
     def run(via, href, args = {}, opts = {})
-      resp = client.send(via, href, args)
+      resp = case via
+             when :get
+               client.send(via, href, args)
+             when :post
+               client.send(via, href) do |req|
+                 req.body = Oj.dump(args)
+               end
+             else
+               raise TeamSnap::Error.new("Don't know how to run `#{via}`")
+             end
 
-      if resp.status == 200
-        collection = Oj.load(resp.body).fetch(:collection)
-        TeamSnap.write_backup_file(opts[:backup_cache_file], collection)
-        collection
+      if resp.success?
+        Oj.load(resp.body).fetch(:collection).tap do |collection|
+          TeamSnap.write_backup_file(opts[:backup_cache_file], collection)
+        end
```

Comment:
style guide says to use `{}` instead of `do...end` here since we want to use the return value of `tap`